source/MRMesh/MRVoxelsConversionsByParts.h

```diff
@@ -0,0 +1,95 @@
+#pragma once
+#if !defined(__EMSCRIPTEN__) && !defined(MRMESH_NO_VOXEL)
+#include "MRMeshFwd.h"
+#include "MRExpected.h"
+#include "MRPartMapping.h"
+
+namespace MR
+{
+
+/**
+ * \struct MR::MergeGridPartSettings
+ * \brief Parameters' structure for MR::mergeGridPart
+ * \ingroup VoxelGroup
+ *
+ * \sa \ref mergeGridPart
+ */
+struct MergeGridPartSettings
+{
+    /// callback to process the generated mesh before the side cutting, e.g. fixing specific generation artifacts
+    using PreCutCallback = std::function<void( Mesh& mesh, float leftCutPosition, float rightCutPosition )>;
+    PreCutCallback preCut = nullptr;
+    /// callback to process the generated mesh after the side cutting, e.g. decimating
+    using PostCutCallback = std::function<void( Mesh& )>;
+    PostCutCallback postCut = nullptr;
+    /// callback to process the destination mesh after merging, usually to map the generated mesh's faces/edges/vertices
+    using PostMergeCallback = std::function<void( Mesh&, const PartMapping& )>;
+    PostMergeCallback postMerge = nullptr;
+    /// mapping with initialized maps required for the `postMerge` callback
+    PartMapping mapping = {};
```

Comment:
do we always get map in callback or only if it is initialized in structure, make it clear